packages/@aws-cdk/aws-apigateway/README.md

```diff
@@ -106,6 +107,31 @@ item.addMethod('GET');   // GET /items/{item}
 item.addMethod('DELETE', new apigateway.HttpIntegration('http://amazon.com'));
 ```
 
+## AWS StepFunctions backed APIs
+
+You can use Amazon API Gateway with AWS Step Functions as the backend integration, specifically Synchronous Express Workflows.
+
+The `StepFunctionsRestApi` construct makes this easy and also sets up input, output and error mapping. The `StepFunctionsRestApi` construct sets up the API Gateway REST API with an `ANY` HTTP method and sets up the api role with the required permission to invoke `StartSyncExecution` action on the AWS StepFunctions state machine. This will enable you to invoke any one of the API Gateway HTTP methods and get a response from the backend AWS StepFunctions state machine.
+
+The following code defines a REST API that routes all requests to the specified AWS StepFunctions state machine:
+
+```ts
+const stateMachine = new stepFunctions.StateMachine(this, 'StateMachine', ...);
+new apigateway.StepFunctionsRestApi(this, 'StepFunctionsRestApi', { 
+  stateMachine: stateMachine,
+});
+```
```

Comment:
I'm still finding it hard to understand.

The paragraph is describing the setup instead it should explain the feature.

Can you improve this example with a simple but actual state machine and explain what invoking the different APIGateway routes would do?

Reply:
We have a pretty nice (unfortunately, undocumented) feature that allows you to insert snippets from an integration test (`integ.*.lit.ts`) directly into the docs.

See https://github.com/aws/aws-cdk/blob/7966f8d48c4bff26beb22856d289f9d0c7e7081d/packages/%40aws-cdk/aws-logs/test/example.retention.lit.ts#L7-L22
and see the README - https://github.com/aws/aws-cdk/blob/7966f8d48c4bff26beb22856d289f9d0c7e7081d/packages/%40aws-cdk/aws-logs/README.md#log-groupsstreams

This finally gets rendered into https://docs.aws.amazon.com/cdk/api/latest/docs/aws-logs-readme.html#log-groupsstreams

It might be worth using that feature here.